### PR TITLE
Introduce a doxygen-checking github action

### PR DESCRIPTION
**Pull request by NlightNFotis**
We used to have one in the Travis days, but somehow we lost
it when we removed the Travis configuration. This PR re-introduces it,
this time working on GitHub actions, along with updating the warnings
we are expecting.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by martin-cs — APPROVED** [review]
My apologies I thought I had fixed all of the doxygen issues in VSD.

**Review by piotr-grabalski — APPROVED** [review]
Looks good to me

File: scripts/expected_doxygen_warnings.txt [hunk 1]
```diff
@@ -1,70 +1,72 @@
...
+/home/runner/work/cbmc/cbmc/src/analyses/reaching_definitions.cpp:679: warning: found </tt> tag without matching <tt>
+/home/runner/work/cbmc/cbmc/src/analyses/reaching_definitions.h:221: warning: The following parameters of rd_range_domaint::merge(const rd_range_domaint &other, locationt from, locationt to) is not documented:
   parameter 'other'
-/cbmc/src/solvers/smt2/smt2_conv.h:113: warning: documented empty return type of  smt2_convt::convert_rounding_mode_FPA
-/cbmc/src/goto-symex/symex_slice_class.h:25: warning: documented empty return type of  symex_slicet::collect_open_variables
+/home/runner/work/cbmc/cbmc/src/solvers/smt2/smt2_conv.h:113: warning: documented empty return type of smt2_convt::convert_rounding_mode_FPA
+/home/runner/work/cbmc/cbmc/src/goto-symex/symex_slice_class.h:25: warning: documented empty return type of symex_slicet::collect_open_variables
+/home/runner/work/cbmc/cbmc/src/goto-instrument/cover_basic_blocks.h:122: warning: argument 'block.' of command @param is not found in the argument list of cover_basic_blockst::add_block_lines(cover_basic_blockst::block_infot &block, const goto_programt::instructiont &instruction)
```
Comment (hannes-steffenhagen-diffblue) [inline scripts/expected_doxygen_warnings.txt:64]: This warning does not seem to appear in the old version. This isn't a problem in itself, but the commit is claiming its only changing path and line numbers.

File: scripts/run_doxygen.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 echo "Running doxygen version $(doxygen --version)"
 
 # Check doxygen version
-EXPECTED_VERSION="1.8.16"
+EXPECTED_VERSION="1.8.17"
```
Comment (hannes-steffenhagen-diffblue) [inline scripts/run_doxygen.sh:6]: This is not a path or line number either.

**Pull request merged**